### PR TITLE
[AppBundle] standardize addresses and geocoding.

### PR DESCRIPTION
**Pull request by hhamon**

### CODE REVIEW

File: src/AppBundle/Address/Address.php [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+ */
+class Address
+{
+    const FRANCE = 'FR';
+
+    /**
+     * @Assert\NotBlank(message="common.address.required")
+     * @Assert\Length(max=150, maxMessage="common.address.max_length")
+     */
+    private $address;
```
Comment (HeahDude) [inline src/AppBundle/Address/Address.php]: street? It's weird to have `address.address` or `$address->getAddress()`.

File: src/AppBundle/Entity/PostAddress.php [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    const FRANCE = 'FR';
+
+    /**
+     * The address street.
+     *
+     * @var string|null
+     *
+     * @ORM\Column(length=150, nullable=true)
+     */
+    private $address;
```
Comment (HeahDude) [inline src/AppBundle/Entity/PostAddress.php]: street?

File: src/AppBundle/Form/CreateCommitteeCommandType.php [hunk 3]
```diff
@@ -52,6 +42,9 @@ public function configureOptions(OptionsResolver $resolver)
     {
         $resolver->setDefaults([
             'data_class' => CommitteeCreationCommand::class,
+            'error_mapping' => [
+                'address.postalCode' => null,
```
Comment (HeahDude) [inline src/AppBundle/Form/CreateCommitteeCommandType.php]: 🤔

File: src/AppBundle/Form/AddressType.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+                'required' => false,
+            ])
+            ->add('cityName', HiddenType::class, [
+                'required' => false,
+            ])
+            ->add('country', UnitedNationsCountryType::class)
+        ;
+    }
+
+    public function configureOptions(OptionsResolver $resolver)
```
Comment (HeahDude) [inline src/AppBundle/Form/AddressType.php]: If data is always passed on form initialization, you can remove this method.

Comment (hhamon) [inline src/AppBundle/Form/AddressType.php]: it's for expliciteness.

Comment (HeahDude) [inline src/AppBundle/Form/AddressType.php]: But will be deprecated soon ;)

Comment (hhamon) [inline src/AppBundle/Form/AddressType.php]: Fuck!

File: src/AppBundle/Committee/CommitteeCreationCommand.php [hunk 5]
```diff
@@ -80,12 +67,13 @@ class CommitteeCreationCommand
...
     public $acceptContactingTerms;
 
-    public function __construct(Adherent $adherent)
+    public function __construct(Adherent $adherent, Address $address = null)
     {
         $this->adherent = $adherent;
         $this->country = 'FR';
         $this->acceptConfidentialityTerms = false;
         $this->acceptContactingTerms = false;
+        $this->address = $address ?? new Address();
```
Comment (HeahDude) [inline src/AppBundle/Committee/CommitteeCreationCommand.php]: `$address ?: new Address()`, no need to test if the variable is set, only if it's not null.

File: src/AppBundle/Committee/CommitteeFactory.php [hunk 6]
```diff
@@ -2,14 +2,22 @@
...
 use AppBundle\Entity\Committee;
 use Ramsey\Uuid\Uuid;
 
 class CommitteeFactory
 {
+    private $addressFactory;
+
+    public function __construct(PostAddressFactory $addressFactory = null)
+    {
+        $this->addressFactory = $addressFactory ?? new PostAddressFactory();
```
Comment (HeahDude) [inline src/AppBundle/Committee/CommitteeFactory.php]: `?:`

File: src/AppBundle/Committee/CommitteeCreationCommand.php (hunk 5 above)
Comment (hhamon) [inline src/AppBundle/Committee/CommitteeCreationCommand.php]: true

File: src/AppBundle/Committee/CommitteeFactory.php (hunk 6 above)
Comment (hhamon) [inline src/AppBundle/Committee/CommitteeFactory.php]: true
